src/unity.c

```diff
@@ -203,19 +188,21 @@ void UnityPrintNumberUnsigned(const UNITY_UINT number)
 /*-----------------------------------------------*/
 void UnityPrintNumberHex(const UNITY_UINT number, const char nibbles_to_print)
 {
-    UNITY_UINT nibble;
+    int nibble;
     char nibbles = nibbles_to_print;
+    if ((unsigned)nibbles > 2 * sizeof number) nibbles = 2 * sizeof number;
```

Comment:
plus, we try to go overboard with parenthesis on this project:

if ((unsigned) nibbles > (2 * sizeof(number)) )
    nibbles = 2 * sizeof(number);